packages/jest-snapshot/src/index.ts

```diff
@@ -122,22 +122,35 @@ const cleanup = (
   hasteFS: HasteFS,
   update: Config.SnapshotUpdateState,
   snapshotResolver: JestSnapshotResolver,
-) => {
+): {
+  filesRemoved: number;
+  filesRemovedList: Array<string>;
+} => {
   const pattern = '\\.' + EXTENSION + '$';
   const files = hasteFS.matchFiles(pattern);
```

Comment:
@ndelangen I think this is the basis of our failing CI. I _think_ this should filter out things from `testPathIgnorePatterns`. We shouldn't care about snapshots for tests which we ignore. Might be wrong since snapshots are not necessarily co-located (if people have custom snapshot serializers), though.

@jeysal @thymikee thoughts?

Reply:
Essentially, I think, when we do `snapshotResolver.resolveTestPath(snapshotFile)`, we should check that the test file is not also ignored, that should fix the issue

Reply:
I'm unsure how to do that.

Reply:
This is on master, so won't apply cleanly, but should work:

```diff
diff --git i/packages/jest-core/src/TestScheduler.ts w/packages/jest-core/src/TestScheduler.ts
index ff29bde93..e3f4f5e81 100644
--- i/packages/jest-core/src/TestScheduler.ts
+++ w/packages/jest-core/src/TestScheduler.ts
@@ -145,6 +145,7 @@ export default class TestScheduler {
           context.hasteFS,
           this._globalConfig.updateSnapshot,
           snapshot.buildSnapshotResolver(context.config),
+          context.config.testPathIgnorePatterns,
         );
 
         aggregatedResults.snapshot.filesRemoved += status.filesRemoved;
diff --git i/packages/jest-snapshot/src/index.ts w/packages/jest-snapshot/src/index.ts
index dd9978d1a..e0439049a 100644
--- i/packages/jest-snapshot/src/index.ts
+++ w/packages/jest-snapshot/src/index.ts
@@ -122,11 +122,24 @@ const cleanup = (
   hasteFS: HasteFS,
   update: Config.SnapshotUpdateState,
   snapshotResolver: JestSnapshotResolver,
+  testPathIgnorePatterns?: Config.ProjectConfig['testPathIgnorePatterns'],
 ) => {
   const pattern = '\\.' + EXTENSION + '$';
   const files = hasteFS.matchFiles(pattern);
+  let testIgnorePatternsRegex: RegExp | null = null;
+  if (testPathIgnorePatterns && testPathIgnorePatterns.length) {
+    testIgnorePatternsRegex = new RegExp(testPathIgnorePatterns.join('|'));
+  }
+
   const filesRemoved = files.reduce((acc, snapshotFile) => {
-    if (!fileExists(snapshotResolver.resolveTestPath(snapshotFile), hasteFS)) {
+    const testPath = snapshotResolver.resolveTestPath(snapshotFile);
+
+    // ignore snapshots of ignored tests
+    if (testIgnorePatternsRegex && testIgnorePatternsRegex.test(testPath)) {
+      return acc;
+    }
+
+    if (!fileExists(testPath, hasteFS)) {
       if (update === 'all') {
         fs.unlinkSync(snapshotFile);
       }
```

Reply:
Potentially, we could _warn_ about this case, but not exit with `1`.

Reply:
added it